posts/inside-rust/2024-08-15-this-development-cycle-in-cargo-1.81.md

```diff
@@ -0,0 +1,426 @@
+Once that was in place, they added building multiple `.crate` files
+([#13947](https://github.com/rust-lang/cargo/pull/13947)).
+
+##### Fingerprinting builds
+
+[Xaeroxe](https://github.com/Xaeroxe) restarted the conversation around fingerprinting source using checksums, instead of mtimes, on
+[zulip](https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Implementing.20checksum.20based.20fingerprinting).
+Specifically, they were starting work on this and wanted to avoid race conditions between when cargo checksummed the source and when rustc built.
+Ideally, rustc would checksum the source to avoid this.
+Some past work in this direction was shared.
+From there, they created a compiler [MCP](https://github.com/rust-lang/compiler-team/issues/765)
+and a [PR](https://github.com/rust-lang/rust/pull/126930) against the compiler.
+However, to do this with `build.rs` would require `build.rs` files to opt-in and they would need to be coordinate on checksum algorithms.
+We decided to punt on this and leave `build.rs` using mtime for now.
+
+[Xaeroxe](https://github.com/Xaeroxe) posted [#14137](https://github.com/rust-lang/cargo/pull/14137) an initial version of checksum fingerprinting for Cargo.
+
+##### `cargo info`
+
+*Update from [1.79](https://blog.rust-lang.org/inside-rust/2024/05/07/this-development-cycle-in-cargo-1.79.html#cargo-info)*
+
+After waiting some time to collect feedback,
+[Rustin170506](https://github.com/Rustin170506)
+proposed `cargo info` to be merged into cargo ([#14141](https://github.com/rust-lang/cargo/pull/14141)).
+This included adapting the code from being written as a third-party to being written in a way Cargo expects, documentation was added, and completion scripts.
+
+There was some discussion on whether to insta-stabilize.
+As a third-party command, we are able to get the testing benefits of nightly-only features.
+If we don't insta-stabilize it, you could only run it as `cargo +nightly -Zsubcommand info` or run the existing `cargo-info` as `cargo info` but `cargo-info info`; `cargo info` would fail.
+Like with `cargo add` and `cargo remove`, we decided to insta-stabilize.
+
+FCP was started for this command.
+
+## Design discussions
+
+##### `--lockfile-path`
+
+When you (or rust-analyzer) runs `cargo metadata`,
+Cargo will ensure the lockfile is in-sync with the `Cargo.toml` file,
+generating it if needed.
+However, if you do this on a read-only filesystem, then it fails
+([#10096](https://github.com/rust-lang/cargo/issues/10096)).
+
+When discussing that problem, the main perspectives were
+
+- `cargo metadata` should be consistent in its output which can only happen if we can write a `Cargo.lock` to the filesystem
+- `cargo metadata` should always let me query information, even it means the information changes due to outside changes (e.g. a new version of a dependency is published so it gets selected)
+
+The problem I see with the latter is user intention.
```

Comment:
Informational, flagging for consideration: this is the only use of "I" in the entire post, AFAICT. Everywhere else uses "we" so it's a bit of a surprising deviation. It isn't entirely clear if this is meant as "the cargo team believes X" or "I, the author, in an individual capacity believe this."